File: ats-ness-back-prod/src/main/java/nesstechnologies/fr/applicationesnbackend/entities/LinkedIn.java
package nesstechnologies.fr.applicationesnbackend.entities;

import jakarta.persistence.*;
import lombok.*;

import java.io.Serializable;
import java.util.List;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Builder
public class LinkedIn implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private String lien;
    private String jobTitle;
    @Lob
    @Column(columnDefinition = "LONGTEXT")
    private  String jobDescription;
    private  String jobLocation;
    private  String previousCompany;
    private  String previousJob;
    @Lob
    @Column(columnDefinition = "LONGTEXT")
    private  String previousJobDescription;
    @ElementCollection
    private List<String> allSkills;
    @ElementCollection
    private List<String> skillsJobActuel;
    @ElementCollection
    private List<String> skillsJobPrecedent;
    private  String libelleGeneral;
    private  int degreActiviteLinkedIn;
    @Lob
    @Column(columnDefinition = "LONGTEXT")
    private  String infos;
    private  String posteActuel;
    private  String localisationLinkedin;
    private  float dureeDerniereSociete;
    @Lob
    @Column(columnDefinition = "LONGTEXT")
    private  String descriptionPosteDerniereSociete;
    @ElementCollection
    private List<String> motsCles;
    @Lob
    @Column(columnDefinition = "LONGTEXT")
    private  String commentaires;
    private  int nbrPushCV;













}

File: ats-ness-front-prod/src/app/demo/models/LinkedIn.ts
export interface LinkedIn {
    id: number;
    lien: string;
    jobTitle: string;
    jobDescription: string;
    jobLocation: string;
    previousCompany: string;
    previousJob: string;
    previousJobDescription: string;
    allSkills: string[];
    skillsJobActuel: string[];
    skillsJobPrecedent: string[];
    libelleGeneral: string;
    degreActiviteLinkedIn: number;
    infos: string;
    posteActuel: string;
    localisationLinkedin: string;
    dureeDerniereSociete: number;
    descriptionPosteDerniereSociete: string;
    motsCles: string[];
    commentaires: string;
    nbrPushCV: number;
}
